### PR TITLE
feat(UPM-7645): support creating base and addon policies

### PR DESCRIPTION
**Pull request by fisher046**

### CODE REVIEW

File: aws/biganimal-csp-setup [hunk 1]
```diff
@@ -326,7 +365,8 @@ EOF
 
     # attach IAM policy to the newly created role
     echo -e "Attaching AWS IAM policy ${GREEN}biganimal-${suffix_policy_name}${NC} to role ${GREEN}biganimal-${suffix_role_name}${NC} within AWS account ${GREEN}${account}${NC}..."
```
Comment (valeriodelsarto) [inline aws/biganimal-csp-setup:367]:
```suggestion
    echo -e "Attaching AWS IAM policy ${GREEN}biganimal-base-${suffix_policy_name}${NC} to role ${GREEN}biganimal-${suffix_role_name}${NC} within AWS account ${GREEN}${account}${NC}..."
```

File: aws/biganimal-csp-setup [hunk 2]
```diff
@@ -326,7 +365,8 @@ EOF
...
-    aws iam attach-role-policy --policy-arn "$policy_arn" --role-name "biganimal-$suffix_role_name"
+    aws iam attach-role-policy --policy-arn "$base_policy_arn" --role-name "biganimal-$suffix_role_name"
+    aws iam attach-role-policy --policy-arn "$addon_policy_arn" --role-name "biganimal-$suffix_role_name"
```
Comment (valeriodelsarto) [inline aws/biganimal-csp-setup:369]:
```suggestion
    echo -e "Attaching AWS IAM policy ${GREEN}biganimal-addon-${suffix_policy_name}${NC} to role ${GREEN}biganimal-${suffix_role_name}${NC} within AWS account ${GREEN}${account}${NC}..."
    aws iam attach-role-policy --policy-arn "$addon_policy_arn" --role-name "biganimal-$suffix_role_name"
```

File: aws/biganimal_AWS_basepolicy_v2.json [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+                "kms:TagResource",
+                "kms:UpdateAlias"
+            ],
+            "Effect": "Allow",
+            "Resource": [
+                "arn:aws:kms:*:*:key/*",
+                "arn:aws:kms:*:*:alias/kms-cmk-*"
+            ],
+            "Sid": "BigAnimalKMSPolicy"
+        },
```
Comment (valeriodelsarto) [inline aws/biganimal_AWS_basepolicy_v2.json:218]: Have we tried to add a `Condition` here like
`
            "Condition": {
                "StringEquals": {
                    "aws:ResourceTag/ManagedBy": "BigAnimal"
                }
            },
`?

**Review by valeriodelsarto — COMMENTED** [review]
added just a few comments. Great work both @fisher046 and @Simonzhaohui here! 🥇

File: aws/biganimal_AWS_basepolicy_v2.json (hunk 3 above)
Comment (fisher046) [reply aws/biganimal_AWS_basepolicy_v2.json:218]: @Simonzhaohui do we have the tag `ManagedBy:BigAnimal` for KMS resources?

**Comment by valeriodelsarto** [comment]
Rebased to the latest main, would you mind @fisher046 and @Simonzhaohui to review it again when you are back? Thanks! 🙏

**Comment by fisher046** [comment]
@valeriodelsarto Your change looks good to me, but I can't approve this PR since it's started by me 😞

**Review by Simonzhaohui — DISMISSED** [review]
LGTM. Thanks

**Review by valeriodelsarto — DISMISSED** [review]
LGTM

**Review by Anhui-tqhuang — APPROVED** [review]

**Pull request merged**